devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,667 @@
+				cmds << "delay 3000"
+				cmds << sensorMultilevelGetCmd(lightSensorType)
+			}
+		}
+	}
+	state.refreshConfig = false
+	return cmds
+}
+
+// Required for HealthCheck Capability, but doesn't actually do anything because this device sleeps.
+def ping() {
+	log.debug "ping()"
+}
+
+// Forces the configuration to be resent to the device the next time it wakes up.
+def refresh() {
+	logForceWakeupMessage "The sensor data will be refreshed the next time the device wakes up."
+	state.lastBatteryReport = null
+	state.lastBattery = null
+	if (!state.refreshSensors) {
+		state.refreshSensors = true
+	} else {
+		state.refreshConfig = true
+	}
+	refreshPendingChanges()
+	return []
+}
+
+private logForceWakeupMessage(msg) {
+	log.debug "${msg}  You can force the device to wake up immediately by holding the z-button for 2 seconds."
+}
+
+def parse(String description) {
+	def result = []
+	try {
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		} else {
+			log.debug "Unable to parse description: $description"
+		}
+
+		sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
+	} catch (e) {
+		log.error "$e"
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapCmd = cmd.encapsulatedCommand(commandClassVersions)
+
+	def result = []
+	if (encapCmd) {
+		result += zwaveEvent(encapCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.wakeupv1.WakeUpNotification cmd) {
+	log.debug "Device Woke Up"
+
+	def cmds = []
+	if (state.refreshConfig || pendingChanges > 0) {
+		cmds += getConfigCmds()
+	}
+
+	if (canReportBattery()) {
+		cmds << batteryGetCmd()
+	}
+
+	if (state.refreshSensors) {
+		cmds += [
+			sensorBinaryGetCmd(),
+			sensorMultilevelGetCmd(tempSensorType),
+			sensorMultilevelGetCmd(lightSensorType)
+		]
+		state.refreshSensors = false
+	}
+
+	if (cmds) {
+		cmds = delayBetween(cmds, 1000)
+		cmds << "delay 3000"
+	}
+	cmds << secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	return response(cmds)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	} else if (val < 1) {
+		val = 1
+	}
+	state.lastBatteryReport = new Date().time
+	state.lastBattery = val
+	log.debug "Battery ${val}%"
+	sendEvent(getEventMap("battery", val, null, null, "%"))
+	return []
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.trace "SensorMultilevelReport: ${cmd}"
+	switch (cmd.sensorType) {
+		case tempSensorType:
+			def unit = cmd.scale ? "F" : "C"
+			def temp = convertTemperatureIfNeeded(cmd.scaledSensorValue, unit, cmd.precision)
+			sendEvent(getEventMap("temperature", temp, true, null, getTemperatureScale()))
+			break
+
```

Comment:
What about removing this empty line break?

Reply:
I remove it.